qcengine/programs/psi4.py

```diff
@@ -93,28 +95,33 @@ def compute(self, input_model: 'ResultInput', config: 'JobConfig') -> 'Result':
                 output_data["error"] = {"error_type": "execution_error", "error_message": output["stderr"]}
 
         else:
-            raise TypeError("Psi4 version '{}' not understood.".format(self.get_version()))
+            raise ResourceError("Psi4 version '{}' not understood.".format(self.get_version()))
 
         # Reset the schema if required
         output_data["schema_name"] = "qcschema_output"
 
         # Dispatch errors, PSIO Errors are not recoverable for future runs
         if output_data["success"] is False:
-
-            if "PSIO Error" in output_data["error"]:
-                raise ValueError(output_data["error"])
+            error_message = output_data["error"]["error_message"]
+
+            if "PSIO Error" in error_message:
+                raise ResourceError(error_message)
```

Comment:
from slack discussion, PSIO is turning into `RandomError`? I agree.

Reply:
Hmm, could be Resource or Random. Do the error strings differ?

Reply:
Apparently PSIO errors should have all the variation one could want (https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libpsio/error.cc). In practice, my mind knows only one PSIO error. So long as it's not on infinite retry, I think a recoverable error is safe and helpful.

Reply:
Attempt to refine this particular error based off those error strings.